src/libs/actions/Policy.ts

```diff
@@ -630,6 +637,65 @@ function removeMembers(accountIDs: number[], policyID: string) {
     API.write(WRITE_COMMANDS.DELETE_MEMBERS_FROM_WORKSPACE, params, {optimisticData, successData, failureData});
 }
 
+function updateWorkspaceMembersRole(policyID: string, accountIDs: number[], newRole: typeof CONST.POLICY.ROLE.ADMIN | typeof CONST.POLICY.ROLE.USER) {
+    const previousPolicyMembers = {...allPolicyMembers};
+    const data: WorkspaceMembersRoleData[] = accountIDs
+        .map((accountID) => {
+            if (!allPersonalDetails?.[accountID]) {
+                return null;
+            }
+
+            return {
+                accountID,
+                email: allPersonalDetails?.[accountID]?.login,
+                role: newRole,
+            };
+        })
+        .filter((item): item is WorkspaceMembersRoleData => item !== null);
```

Comment:
We could use `reduce` to push  create data so that we don't need to call filter:

```suggestion
        .reduce((accountID, result) => {
            if (!allPersonalDetails?.[accountID]) {
                return result;
            }

            result.push({
                accountID,
                email: allPersonalDetails?.[accountID]?.login,
                role: newRole,
            });
        }, []);
```

Reply:
done